lib/systemd_testsuite_test.pm

```diff
@@ -53,7 +53,7 @@ sub testsuiteinstall {
             wait_still_screen 10;
             send_key 'ret';
         }
-        wait_serial('Welcome to', 300) || die "System did not boot in 300 seconds.";
+        assert_screen('linux-login', 30);
```

Comment:
The problem with the needle is that each time that the font change (it happens too often) a new needle need to be created, creating a lot of extra work. If the needle is created with too little match %, then it probably match when it shouldn't, making the test to fail.

Now, there is something I don't understand. This is only checked if BOOT_HDD_IMAGE, but the login prompt should be checked by the previous module boot_to_desktop. This block should be removed completely. Do you know an scenario where this block is needed?

```perl
if (get_var('BOOT_HDD_IMAGE')) {
    if (check_var('ARCH', 'aarch64')) {
        wait_still_screen 10;
        send_key 'ret';
    }
    wait_serial('Welcome to', 300) || die "System did not boot in 300 seconds.";
}
```

Reply:
> Now, there is something I don't understand. This is only checked if BOOT_HDD_IMAGE, but the login prompt should be checked by the previous module boot_to_desktop.

The login prompt is actually being checked already in the module `boot_to_desktop` -> `opensusebasetest::wait_boot` -> `opensusebasetest::wait_boot_textmode` -> `testapi::assert_screen`. 

> This block should be removed completely. Do you know an scenario where this block is needed?

I agree, as far as I can tell it can be safely completely removed.

Reply:
I've had a problem on aarch64 that the system got stuck in the bootloader, e.g. there was no bootloader timeout. Therefore I've added the send_key 'ret' to confirm the choosen bootloader entry. Afterwards I had to add the wait_serial for the system to boot up. That hack might not be necessary in the current testcode anymore. Sergio, maybe you could double ckeck on an aarch64 testmachine?

Reply:
The schedule for aarch64 looks wrong: https://openqa.opensuse.org/tests/1203476

Reply:
I think it is time to use the YAML schedule for this test suite. I will propose a PR later, when I finish some tasks.

But, this PR will not cause any problem since the test suite for aarch64 is already failing:
- openSUSE: https://openqa.opensuse.org/tests/1203476
- SLE: https://openqa.suse.de/tests/3994532#next_previous